Makefile

```diff
@@ -0,0 +1,143 @@
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
+
+## Tool Versions
+KUSTOMIZE_VERSION ?= v3.8.7
+
+OUTPUT_DIR ?= $(PROJECT_DIR)/out
+$(OUTPUT_DIR):
+	mkdir -p $(OUTPUT_DIR)
+PYTHON_VENV_DIR ?= $(OUTPUT_DIR)/venv3
+HACK_DIR ?= $(PROJECT_DIR)/hack
+SCRIPTS_DIR = $(PROJECT_DIR)/scripts
+TEMP_DIR = $(PROJECT_DIR)/temp
+
+PRIMAZA_CONFIG_DIR ?= $(SCRIPTS_DIR)/config
+$(PRIMAZA_CONFIG_DIR):
+	mkdir -p $(PRIMAZA_CONFIG_DIR)
+
+PRIMAZA_CONFIG_FILE = $(PRIMAZA_CONFIG_DIR)/primaza_config_latest.yaml
+
+KIND_CONFIG_DIR ?= $(SCRIPTS_DIR)/src/primazatest/config
+KIND_CONFIG_FILE ?= $(KIND_CONFIG_DIR)/kind.yaml
+KIND_CLUSTER_NAME ?= primazactl-test
+KUBE_KIND_CLUSTER_NAME ?= kind-$(KIND_CLUSTER_NAME)
+
+KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+.PHONY: kustomize
+kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
+$(KUSTOMIZE): $(LOCALBIN)
+	test -s $(LOCALBIN)/kustomize || { curl -Ss $(KUSTOMIZE_INSTALL_SCRIPT) | bash -s -- $(subst v,,$(KUSTOMIZE_VERSION)) $(LOCALBIN); }
+
+.PHONY: config
+config: clone kustomize $(PRIMAZA_CONFIG_DIR) ## Get config files from primaza repo.
+	-rm $(PRIMAZA_CONFIG_FILE)
+	cd $(TEMP_DIR)/config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
+	$(KUSTOMIZE) build $(TEMP_DIR)/config/default > $(PRIMAZA_CONFIG_FILE)
+
+.PHONY: primaza-main-controllers
+primaza-main-controllers: clone
+	cd temp && export IMG=$(IMG) && make primaza docker-build
+	cd temp && export IMG=$(IMG) && make primaza docker-push
+
+ifeq ($(shell sh -c 'uname 2>/dev/null || echo Unknown'),Darwin)
+SED_EXTRA := .bak
+endif
+
+.PHONY: kind-cluster
+kind-cluster: $(SED-EXTRA)
+	-kind delete cluster --name $(KIND_CLUSTER_NAME)
+	##kind create cluster --name $(KIND_CLUSTER_NAME) && kubectl wait --for condition=Ready nodes --all --timeout=600s
+	sed -i $(SED_EXTRA) 's/name:.*/name: $(KIND_CLUSTER_NAME)/g' $(KIND_CONFIG_FILE)
+	kind create cluster --config $(KIND_CONFIG_FILE) && kubectl wait --for condition=Ready nodes --all --timeout=600s
+
+.PHONY: setup-test
+setup-test: clean kind-cluster primazactl config
+
+.PHONY: clone
+clone: clean-temp
+	git clone $(PRIMAZA_REPO) $(TEMP_DIR)
+	cd $(TEMP_DIR) && git checkout $(PRIMAZA_BRANCH)
+
+.PHONY: primazactl
+primazactl: ## Setup virtual environment
+	-rm -rf $(PYTHON_VENV_DIR)
+	python3 -m venv $(PYTHON_VENV_DIR)
+	$(PYTHON_VENV_DIR)/bin/pip install --upgrade setuptools
+	$(PYTHON_VENV_DIR)/bin/pip install --upgrade pip
+	cd $(SCRIPTS_DIR) && $(PYTHON_VENV_DIR)/bin/pip3 install -r requirements.txt
+	cd $(SCRIPTS_DIR) && $(PYTHON_VENV_DIR)/bin/python3 setup.py install
+
+.PHONY: lint
+lint: primazactl ## Check python code
+	PYTHON_VENV_DIR=$(PYTHON_VENV_DIR) $(HACK_DIR)/check-python/lint-python-code.sh
+
+.PHONY: test
+test: setup-test
+	$(PYTHON_VENV_DIR)/bin/primazatest -v $(PYTHON_VENV_DIR) -f $(PRIMAZA_CONFIG_FILE) -c $(KUBE_KIND_CLUSTER_NAME)
+
+.PHONY: clean-temp
+clean-temp:
+	-chmod 755 $(TEMP_DIR)/bin/k8s/*
```

Comment:
I'd mark this as an argument to get rid of `envtest` in primaza; I don't think we use it all that much, and it's rather annoying to deal with its artifacts.

Reply:
when you get rid of it let me know.

Reply:
I'll file a PR for this change on primaza, but I don't want that to block this PR.  When it lands, I'll make a separate PR for it here on primazactl.